src/Coin.cpp

```diff
@@ -0,0 +1,264 @@
+
+using namespace TW;
+
+std::string TW::loadAddress(TWCoinType coin, const Data& data) {
+    switch (coin) {
+    case TWCoinTypeBinance:
+        return Tendermint::Address(HRP_BINANCE, data).string();
+
+    case TWCoinTypeBitcoin:
+        return Bitcoin::Bech32Address(HRP_BITCOIN, 0, data).string();
+
+    case TWCoinTypeBitcoinCash:
+        return Bitcoin::CashAddress(data).string();
+
+    case TWCoinTypeCallisto:
+    case TWCoinTypeEthereum:
+    case TWCoinTypeEthereumClassic:
+    case TWCoinTypeGo:
+    case TWCoinTypePoa:
+    case TWCoinTypeThunderToken:
+    case TWCoinTypeTomoChain:
+    case TWCoinTypeVeChain:
+    case TWCoinTypeWanChain:
+        return Ethereum::Address(data).string();
+
+    case TWCoinTypeICON:
+        return Icon::Address(data, TWIconAddressTypeAddress).string();
+
+    case TWCoinTypeLitecoin:
+        return Bitcoin::Bech32Address(HRP_LITECOIN, 0, data).string();
+
+    case TWCoinTypeNimiq:
+        return Nimiq::Address(data).string();
+
+    case TWCoinTypeRipple:
+        return Ripple::Address(data).string();
+
+    case TWCoinTypeDash:
+    case TWCoinTypeTron:
+    case TWCoinTypeZcoin:
+        return Bitcoin::Address(data).string();
+
+    case TWCoinTypeZcash:
+        return Zcash::TAddress(data).string();
+
+    case TWCoinTypeTezos:
+    case TWCoinTypeEOS:
+        return "";
+    }
+}
+
+bool TW::validateAddress(TWCoinType coin, const std::string& string) {
+    switch (coin) {
+    case TWCoinTypeBinance:
+        return Tendermint::Address::isValid(string, HRP_BINANCE) || Tendermint::Address::isValid(string, HRP_BINANCE_TEST);
+
+    case TWCoinTypeBitcoin:
+        return Bitcoin::Bech32Address::isValid(string, HRP_BITCOIN) || Bitcoin::Address::isValid(string, {TWP2PKHPrefixBitcoin, TWP2SHPrefixBitcoin});
+
+    case TWCoinTypeBitcoinCash:
+        return Bitcoin::CashAddress::isValid(string) || Bitcoin::Address::isValid(string, {TWP2PKHPrefixBitcoin, TWP2SHPrefixBitcoin});
+
+    case TWCoinTypeCallisto:
+    case TWCoinTypeEthereum:
+    case TWCoinTypeEthereumClassic:
+    case TWCoinTypeGo:
+    case TWCoinTypePoa:
+    case TWCoinTypeThunderToken:
+    case TWCoinTypeTomoChain:
+    case TWCoinTypeVeChain:
+    case TWCoinTypeWanChain:
+        return Ethereum::Address::isValid(string);
+
+    case TWCoinTypeICON:
+        return Icon::Address::isValid(string);
+
+    case TWCoinTypeLitecoin:
+        return Bitcoin::Bech32Address::isValid(string, HRP_LITECOIN) || Bitcoin::Address::isValid(string, {TWP2PKHPrefixLitecoin, TWP2SHPrefixLitecoin});
+
+    case TWCoinTypeNimiq:
+        return Nimiq::Address::isValid(string);
+
+    case TWCoinTypeRipple:
+        return Ripple::Address::isValid(string);
+
+    case TWCoinTypeDash:
+        return Bitcoin::Address::isValid(string, {TWP2PKHPrefixDash, TWP2SHPrefixDash});
+
+    case TWCoinTypeTezos:
+        return Tezos::Address::isValid(string);
+
+    case TWCoinTypeTron:
+        return Bitcoin::Address::isValid(string, {0x41});
```

Comment:
Use a const variable for `0x41`?

Reply:
Not sure what I should call this. It should be in Tron::Address once we have one.

Reply:
Just tron address prefix?